cpp/src/arrow/type.cc

```diff
@@ -153,8 +153,9 @@ std::string ListType::ToString() const {
 
 MapType::MapType(const std::shared_ptr<DataType>& key_type,
                  const std::shared_ptr<DataType>& item_type, bool keys_sorted)
-    : ListType(struct_({std::make_shared<Field>("key", key_type, false),
-                        std::make_shared<Field>("item", item_type)})),
+    : ListType(std::make_shared<Field>("$data$",
+               struct_({std::make_shared<Field>("key", key_type, false),
+                        std::make_shared<Field>("item", item_type)}), false)),
```

Comment:
Can we call this "value" instead of "item"?

Reply:
sure

Reply:
@wesm The reason for "item" is that "value" is already used in the scope of ListArray; it refers to what will now be called entries. So after the renaming above `map_array.values()` returns the entries and `map_array.items()` returns the values. If we override `map_array.values()` to do the thing everyone will expect, then it's no longer safe to treat ListArrays as MapArrays. Honestly what I'd prefer is to also rename ListArray::values to ListArray::entries or items so we can maintain sane naming

Reply:
@BryanCutler

Reply:
Added a JIRA to track this for later https://issues.apache.org/jira/browse/ARROW-5745

Reply:
@bkietz so is the naming here ok to be merged for now and then we can finalize it in ARROW-5745 as a followup?

Reply:
@BryanCutler I think so, thanks